editor/src/components/canvas/canvas-strategies/strategies/draw-to-insert-metastrategy.tsx

```diff
@@ -48,22 +51,88 @@ import {
   strategyApplicationResult,
   targetPaths,
 } from '../canvas-strategy-types'
-import { boundingArea, InteractionSession } from '../interaction-state'
-import { getReparentTargetUnified, newReparentSubjects } from './reparent-strategy-helpers'
+import { boundingArea, InteractionSession, MissingBoundsHandling } from '../interaction-state'
+import { getApplicableReparentFactories } from './reparent-metastrategy'
+import { ReparentStrategy } from './reparent-strategy-helpers'
 
-export function drawToInsertStrategy(
+export const drawToInsertMetaStrategy: MetaCanvasStrategy = (
   canvasState: InteractionCanvasState,
   interactionSession: InteractionSession | null,
   customStrategyState: CustomStrategyState,
+): Array<CanvasStrategy> => {
+  if (
+    interactionSession == null ||
+    !(
+      interactionSession.interactionData.type === 'DRAG' ||
+      interactionSession.interactionData.type === 'HOVER'
+    )
+  ) {
+    return []
+  }
+  const pointOnCanvas =
+    interactionSession.interactionData.type === 'DRAG'
+      ? interactionSession.interactionData.originalDragStart
+      : interactionSession.interactionData.point
+
+  const cmdPressed = interactionSession.interactionData.modifiers.cmd
+
+  const applicableReparentFactories = getApplicableReparentFactories(
+    canvasState,
+    pointOnCanvas,
+    cmdPressed,
+    true, // <- TODO this is an important assumption, make sure this is still true when inserting into a flex storyboard
```

Comment:
Sounds like a case for a test?

Reply:
now that you mentioned it... there's a ton of things here that should be tested - this PR unlocks a bunch of new behavior!!!